lib/failover_ip.rb

```diff
@@ -70,18 +78,25 @@ def switch_ips
 
       old_target = current_target
 
-      Hooks.run_before failover_ip, old_target, new_ip[:target]
+      Hooks.run_before failover_ip, old_target, new_ip[:target], dry
 
-      raise unless HTTParty.post("#{base_url}/failover/#{failover_ip}", :body => { :active_server_ip => new_ip[:target] }, :basic_auth => basic_auth).success?
+      if !dry
+        response = HTTParty.post("#{base_url}/failover/#{failover_ip}", :body => { :active_server_ip => new_ip[:target] }, :basic_auth => basic_auth)
+        raise unless response.success?
+        $logger.info "Switch #{failover_ip} to #{new_ip[:target]} completed"
+      else
+        $logger.info "Dry run: would have switched #{failover_ip} to #{new_ip[:target]}"
+      end
 
-      Hooks.run_after failover_ip, old_target, new_ip[:target]
+      Hooks.run_after failover_ip, old_target, new_ip[:target], dry
 
       return true
     end
 
     false
   rescue
-    $logger.error "Unable to set a new active server ip for #{failover_ip}"
+    $logger.error "Unable to set #{new_ip[:target]} as new active server ip for #{failover_ip} via POST to #{base_url}/failover/#{failover_ip}, with :basic_auth => #{basic_auth}"
```

Comment:
what about the concern regarding having credentials in the log?

Reply:
I think this has been addressed in 53f0194